docs/content/en/docs/implementing/tasks-non-k8s-apps.md

```diff
@@ -35,15 +58,32 @@ spec:
       - 'hello world'
 ```
 
-## Step 2: Create a KeptnTask for each run
+This example uses the `container-runtime` runner
+because it allows the most flexibility
+but you can also use the `deno-runtime` or `python-runtime` runners.
 
-Each time you want to execute a `KeptnTaskDefinition`, a new (and uniquely named) `KeptnTask` must be created.
-
-In the standard operating mode, when Keptn is managing workloads, the creation of the `KeptnTask` CR is automatic.
+## Create and apply a KeptnTask
 
+You must manually create the
+[KeptnTask](../yaml-crd-ref/task.md)
+the resource.
+In the standard operating mode, when Keptn is managing workloads,
+the creation of the `KeptnTask` resource is automatic.
 Here though, we must create it ourselves.
 
-The `KeptnTask` references the `KeptnTaskDefinition` in the `spec.taskDefinition` field:
+Moreover, you must create a new (and uniquely named)
+`KeptnTask` resource
+each time you want to rerun this task
+Each time you want to execute a `KeptnTask` resource,
+you must manually create a
+a new (and uniquely named)
+[KeptnTask](../yaml-crd-ref/task.md)
+YAML file to describe that resource.
```

Comment:
This is wrong. It's not the YAML filename on disk that matters but the `metadata.name` field which must be unique. `kubectl apply` applies the _contents_ of the YAML file. Kubernetes never knows about the filename

Reply:
This was corrected in the process of updating the syntax of this CRD